src/cli/cli.cpp

```diff
@@ -52,8 +52,8 @@
 #include <common/encoding.hpp>
 #include <common/new.hpp>
 #include <net/ip6.hpp>
-#include <platform/random.h>
 #include <platform/uart.h>
+#include <utils/global_address.hpp>
```

Comment:
Why does CLI need to include this header? I'd prefer that we design such that any OpenThread client shouldn't need to include anything from the /src/core/* path. Either remove the include, move the header to /include or move what ever you need from the header to openthread.h or openthread-types.h?